modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -419,6 +421,12 @@ public static void deleteSnapshot(File snpDir, String folderName) {
                 }
             });
 
+            File binMetadataDfltDir = new File(snpDir, DFLT_BINARY_METADATA_PATH);
+            File marshallerDfltDir = new File(snpDir, DFLT_MARSHALLER_PATH);
+
+            U.delete(binMetadataDfltDir);
+            U.delete(marshallerDfltDir);
```

Comment:
It's not correct in doing such a way since the marshaller directory is shared between all nodes (if nodes are started on the local machine). Thus you will give the `NoSuchFileException` due to `U.delete` doesn't work with directories in concurrent-way.

Reply:
Do you mean it's necessary to check if directory exists prior to deleting it? It's just that I looked inside `U.delete` and it handles absence of files/directories by catching all `IOExceptions`.